crates/flox-rust-sdk/src/models/environment.rs

```diff
@@ -259,10 +263,24 @@ where
     }
 
     /// Activate this environment
-    /// TODO: remove this `allow` once the method is filled out
-    #[allow(unused_variables)]
-    async fn activate(&self, nix: &NixCommandLine) -> Result<(), EnvironmentError2> {
-        todo!()
+    async fn activate(
+        &self,
+        nix: &NixCommandLine,
+        system: impl AsRef<str> + Send,
+    ) -> Result<(), EnvironmentError2> {
+        let command = Shell {
+            eval: EvaluationArgs {
+                impure: true.into(),
+                ..Default::default()
+            },
+            installables: [self.flake_attribute(&system).into()].into(),
+            ..Default::default()
+        };
+        command
+            .run(nix, &Default::default())
+            .await
+            .map_err(EnvironmentError2::Activate)?;
```

Comment:
As this is starting a subshell I think I would expect that it's in the flox crate? I never felt fully resolved about https://github.com/flox/flox/commit/88f147c1783e0b0eefcc7d7a20ab4691a2c68c0c#r124910472

Reply:
How do you expect that to look? The `activate` method is on the `Environment` trait, which lives in `flox-rust-sdk`. I think it makes less sense to have the interface for an environment split across multiple crates. Given that we've already decided that this is a temporary solution, I'd prefer to just stick with this until we have the actual `activate` implementation figured out.

Reply:
If you're going to have a (self-imposed) boundary inside the flox application, I think it makes more sense for flox to be the coordinator and the interface, calling high level methods on the lower level components. It feels arbitrary to me to decide that nothing outside of flox can create a shell.

I'm not trying to pick another fight here, I'm just pointing out that where we decide to put this boundary has implications for simplicity of the interfaces and how natural they feel.